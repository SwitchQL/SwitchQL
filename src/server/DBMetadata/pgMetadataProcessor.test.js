import processed from "./sampleFiles/processedMetadata";
import retrieved from "./sampleFiles/retrievedMetadata";

import processMetadata from "./pgMetadataProcessor";

describe("Format Metadata Tests", () => {
  it("Should return correctly formatted metadata given sample input", () => {
    const result = processMetadata(retrieved);
    expect(result).toEqual(processed);
  });

  it("Throws an error on null or empty data", () => {
    const metadataTest = () => {
      processMetadata([]);
    };

    expect(metadataTest).toThrowError("Metadata is null or empty");
  });

  it("Throws an error on non array input", () => {
    const metadataTest = () => {
      processMetadata({});
    };

    expect(metadataTest).toThrowError(
      "Invalid data format. Column Data must be an array"
    );
  });
});
