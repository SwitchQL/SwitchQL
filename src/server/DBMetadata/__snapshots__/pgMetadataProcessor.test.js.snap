// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format Metadata Tests Should match the snapshot 1`] = `
Object {
  "tables": Object {
    "0": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "2.5": Object {
              "refField": 5,
              "refTable": 2,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 0,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "name",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 0,
          "type": "String",
          "unique": false,
        },
      ],
      "name": "author",
    },
    "1": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": false,
          "name": "id",
          "primaryKey": true,
          "relation": Object {},
          "required": true,
          "tableNum": 1,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": true,
          "name": "book_id",
          "primaryKey": false,
          "relation": Object {
            "2.1": Object {
              "refField": 1,
              "refTable": 2,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 1,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 2,
          "inRelationship": true,
          "name": "order_id",
          "primaryKey": false,
          "relation": Object {
            "4.0": Object {
              "refField": 0,
              "refTable": 4,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 1,
          "type": "ID",
          "unique": false,
        },
      ],
      "name": "book_order",
    },
    "2": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "genre_id",
          "primaryKey": false,
          "relation": Object {
            "3.0": Object {
              "refField": 0,
              "refTable": 3,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 2,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "1.1": Object {
              "refField": 1,
              "refTable": 1,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 2,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 2,
          "inRelationship": false,
          "name": "test",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 2,
          "type": "Float",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 3,
          "inRelationship": false,
          "name": "name",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 2,
          "type": "String",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 4,
          "inRelationship": false,
          "name": "publish_date",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 2,
          "type": "Date",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 5,
          "inRelationship": true,
          "name": "author_id",
          "primaryKey": false,
          "relation": Object {
            "0.0": Object {
              "refField": 0,
              "refTable": 0,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 2,
          "type": "ID",
          "unique": false,
        },
      ],
      "name": "books",
    },
    "3": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "2.0": Object {
              "refField": 0,
              "refTable": 2,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 3,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "name",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 3,
          "type": "String",
          "unique": false,
        },
      ],
      "name": "genre",
    },
    "4": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "1.2": Object {
              "refField": 2,
              "refTable": 1,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 4,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "created_at",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 4,
          "type": "Date",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 2,
          "inRelationship": true,
          "name": "user_id",
          "primaryKey": false,
          "relation": Object {
            "7.0": Object {
              "refField": 0,
              "refTable": 7,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 4,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 3,
          "inRelationship": true,
          "name": "status_id",
          "primaryKey": false,
          "relation": Object {
            "6.0": Object {
              "refField": 0,
              "refTable": 6,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 4,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 4,
          "inRelationship": true,
          "name": "shipping_id",
          "primaryKey": false,
          "relation": Object {
            "5.0": Object {
              "refField": 0,
              "refTable": 5,
              "refType": "many to one",
            },
          },
          "required": false,
          "tableNum": 4,
          "type": "ID",
          "unique": false,
        },
      ],
      "name": "order",
    },
    "5": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "4.4": Object {
              "refField": 4,
              "refTable": 4,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 5,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "method",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 5,
          "type": "String",
          "unique": false,
        },
      ],
      "name": "shipping_method",
    },
    "6": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "4.3": Object {
              "refField": 3,
              "refTable": 4,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 6,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "code",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 6,
          "type": "String",
          "unique": false,
        },
      ],
      "name": "status",
    },
    "7": ProcessedTable {
      "fields": Array [
        ProcessedField {
          "fieldNum": 0,
          "inRelationship": true,
          "name": "id",
          "primaryKey": true,
          "relation": Object {
            "4.2": Object {
              "refField": 2,
              "refTable": 4,
              "refType": "one to many",
            },
          },
          "required": true,
          "tableNum": 7,
          "type": "ID",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 1,
          "inRelationship": false,
          "name": "phone_number",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 7,
          "type": "String",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 2,
          "inRelationship": false,
          "name": "address",
          "primaryKey": false,
          "relation": Object {},
          "required": false,
          "tableNum": 7,
          "type": "String",
          "unique": false,
        },
        ProcessedField {
          "fieldNum": 3,
          "inRelationship": false,
          "name": "name",
          "primaryKey": false,
          "relation": Object {},
          "required": true,
          "tableNum": 7,
          "type": "String",
          "unique": false,
        },
      ],
      "name": "user",
    },
  },
}
`;
